tests/smoke/features/step_definitions/app.files.ts

```diff
@@ -178,3 +175,196 @@ When(
+
+    await allFilesPage.navigate()
+
+    for (const resource of resources) {
+      await allFilesPage.resourceExist({ name: resource })
+    }
+  }
+)
+
+Then(
+  '{string} ensure that resource {string} has {int} versions',
+  async function (this: World, stepUser: string, resource: string, countOfVersion: number) {
+    const actor = this.actorContinent.get({ id: stepUser })
+    const { allFiles: allFilesPage } = new FilesPage({ actor })
+
+    // skipped in Oc10, since the version number in Oc10 is no more than 1
+    if (config.ocis) {
+      await expect(await allFilesPage.numberOfVersions({ resource: resource })).toEqual(
+        countOfVersion
+      )
+    }
+    await allFilesPage.navigate()
+  }
+)
+
+When(
+  '{string} deletes following resource(s)',
+  async function (this: World, stepUser: string, stepTable: DataTable) {
+    const actor = this.actorContinent.get({ id: stepUser })
+    const { allFiles: allFilesPage } = new FilesPage({ actor })
+    const resources = stepTable.raw().map((f) => f[0])
+
+    for (const resource of resources) {
+      await allFilesPage.deleteResourses({ resource: resource })
+    }
+  }
+)
+
+When(
+  '{string} shares following resource(s) using main menu',
+  async function (this: World, stepUser: string, stepTable: DataTable) {
+    const actor = this.actorContinent.get({ id: stepUser })
+    const { allFiles: allFilesPage } = new FilesPage({ actor })
+
+    const shareInfo = stepTable.hashes().reduce((acc, stepRow) => {
+      const { user, resource, role } = stepRow
+
+      if (!acc[resource]) {
+        acc[resource] = { users: [], role: '' }
+      }
+
+      acc[resource].users.push(this.userContinent.get({ id: user }))
+      acc[resource].role = role
+
+      return acc
+    }, {})
+
+    for (const folder of Object.keys(shareInfo)) {
+      await allFilesPage.shareFolder({
+        folder,
+        users: shareInfo[folder].users,
+        role: shareInfo[folder].role,
+        mainMenu: true
+      })
+    }
+  }
+)
+
+When(
+  '{string} opens file in Mediaviewer',
+  async function (this: World, stepUser: string, stepTable: DataTable) {
+    const actor = this.actorContinent.get({ id: stepUser })
+    const { allFiles: allFilesPage } = new FilesPage({ actor })
+    const resource = stepTable.raw().map((f) => f[0])
+
+    await allFilesPage.openFileInMediaviewer({ resource: resource[0] })
+  }
+)
+
+When(
+  '{string} changes role for the following shared resources(s)',
+  async function (this: World, stepUser: string, stepTable: DataTable) {
+    const actor = this.actorContinent.get({ id: stepUser })
+    const { allFiles: allFilesPage } = new FilesPage({ actor })
+
+    const shareInfo = stepTable.hashes().reduce((acc, stepRow) => {
+      const { user, resource, role } = stepRow
+
+      if (!acc[resource]) {
+        acc[resource] = { users: [], role: '' }
+      }
+
+      acc[resource].users.push(this.userContinent.get({ id: user }))
```

Comment:
```suggestion
      acc[resource].users.push(this.userContent.get({ id: user }))
```

Reply:
userContinent is fine https://github.com/owncloud/web/blob/shareFile/tests/smoke/support/world/user.ts